nav2_bringup/bringup/launch/nav2_bringup_launch.py

```diff
@@ -111,21 +109,6 @@ def generate_launch_description():
                               'use_lifecycle_mgr': 'false',
                               'use_remappings': use_remappings,
                               'map_subscribe_transient_local': 'true'}.items()),
-
```

Comment:
This change will result in two instances of lifecycle_manager getting launched, one managing amcl and map server and the other managing the rest of the nodes.

Both lifecycle_managers will have the same node name and be in the same namespace, so they will not be individually addressable. In the case of `autostart=False`, there is no way to reliably address the specific lifecycle managers to cause them to start lifecycle transitions.

The `nav2_tb3_simulation_launch.py` starts the system with `autostart=False` and relies on the RViz plugin to perform the lifecycle transitions. Also the RViz plugin is unaware that it needs to talk to two lifecycle managers now.

There are 3 approaches I can see to address this problem:
1. Leave it as it was
1. Put the two lifecycle managers in separate namespaces, always autostart, and remove lifecycle_manager interaction from the Rviz plugin.
3. Put the two lifecycle managers in separate namespaces, and add code to find all lifecycle_manager services in the system, and start them all when the startup button is pressed.
4. Change the way lifecycle manager works so each launch file can add it's own nodes to the startup list.

Reply:
Just going inline:

> This change will result in two instances of lifecycle_manager getting launched, one managing amcl and map server and the other managing the rest of the nodes.

Agreed.

> Both lifecycle_managers will have the same node name and be in the same namespace, so they will not be individually addressable. 

That is not true, they have different names, `lifecycle_manager_navigation` and `lifecycle_manager_localization`.

> In the case of autostart=False, there is no way to reliably address the specific lifecycle managers to cause them to start lifecycle transitions.

That would be true, I hadn't thought about that. 

If these are truly intended to be independent launch files, I think its totally reasonable that they each have their own lifecycle manager, or we need a single global-scope one that is given all the nodes to launch. The benefit of this is that a single launch file contains everything it needs to be successful. If you want to run your navigation with mapping, kill mapping, and move to localization, you can do that without effecting the navigation's lifecycle. That's very beneficial to me. I wouldn't like the idea of a navigation-base launch file without being able to transition fully into active. I think you could optionally dump the AMCL/map server stuff into that same lifecycle manager, but its not clear to me how to do that in the current launch system.

If we chose to leave it as it was, then I think the ticket should be closed, because that is the scope of the ticket, to remove redundancy in the lifecycle node declarations. I'd prefer to default everything to auto-bringup or add it to the rviz plugin, but that doesn't seem like a good move to have multiple, possibly there, possibly not lifecycle manager clients depending on if you're in slam or localization mode. 

Thoughts?

Reply:
Another point to mention, those other 2 lifecycle managers were inside the launch files, I just made them default and removed the single one. If the rviz plugin or multiple managers is an issue, those then need to just be flat out removed.

Reply:
> I think its totally reasonable that they each have their own lifecycle manager

I'm ok with that, but there was some resistance to having two separate lifecycle managers running, back when these launch files were rewritten.  @mkhansen-intel Were you the one who preferred have one lifecycle manager running instead of two?

If we go the route of having two lifecycle managers, I'm ok with making autostart=True the default in all our launch files. For the Rviz plugin, I'm inclined to pull the lifecycle stuff out of it and making the lifecycle management a separate tool

Reply:
Or, we just hardcode in lifecycle managers startup and just entirely abstract out the concept to users. 

I don't mind the button, but it does make it so that any modification to the example launch files basically makes the rviz plugin stop functioning, which isn't very bulletproof.

More thoughts would be good on this before going one way or another. Maybe a discussion point for tomorrow.

Reply:
I think we're on the same page. I was thinking the separate lifecycle management tool would be outside of RViz and completely optional if the user wants to use our system lifecycle transitions.

I think the long-term goal is to migrate lifecycle manager into `launch`. This seems like a step in that direction.

Reply:
Well Launch has a way to transition into lifecycle active states. Its a mess, but it exists. Though I haven't figured out how to transition to shutdown on control-C, but I'm not convinced its _not_ possible, but challenging. Long-term reimplementing the lifecycle manager as a python launch system item sounds interesting. Maybe the `LifecycleNode()` has another flag for `add_to_manager` and in the background it aggregates the _ordered_ lifecycle nodes and brings them all up in that fashion. Still need to figure out how to handle the `SIG`s to transition to shutdown before exiting. This sounds like a really good future work item. I'll file a ticket as a reminder. Maybe I'll have an intern eventually who can do that.

The rviz plugin panel really only deals with the lifecycle states and pausing navigation. I think that's an appropriate location for it, though I'd recommend automatically starting up by default in all launch files. The less friction for users to get this going the better. Plus if they dont want to use rviz, that will make it very hard for them to find out about the activation stuff (well, I guess other than the documentation ;-) ).